src/pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -243,8 +246,8 @@ def _reorder_callbacks(callbacks: List[Callback]) -> List[Callback]:
             A new list in which the last elements are Checkpoint if there were any present in the
             input.
         """
-        checkpoints = [c for c in callbacks if isinstance(c, Checkpoint)]
-        not_checkpoints = [c for c in callbacks if not isinstance(c, Checkpoint)]
+        checkpoints = [c for c in callbacks if isinstance(c, Checkpoint and Callback)]
```

Comment:
First of all, this is not how you check for multiple types - and we don't want to check for callback.

```suggestion
        checkpoints: List[Callback] = [c for c in callbacks if isinstance(c, Checkpoint)]
```